@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+        const endIdx = featureIdx < lastFeatureIdx ? featureStarts[++featureIdx] : groupEndIdx;
+        if (bVolumeIdx >= bVolumes.length) {
+            bVolumes.push(computeFeatureBoundingSphere(geometry, beginIdx, endIdx));
+        }
+        intersectFeature(
+            mesh,
+            raycaster,
+            localRay,
+            intersects,
+            halfWidth,
+            localHalfWidth,
+            localHalfWidthSq,
+            beginIdx,
+            endIdx,
+            bVolumes[bVolumeIdx]
+        );
+        beginIdx = endIdx;
+    }
+    return featureIdx;
+}
+
+/**
+ * Mesh formed by extruding a polyline in the shaders. Overrides raycasting behaviour to account for
+ * extrusion, see [[SolidLineMaterial]].
+ * @internal
+ */
+export class SolidLineMesh extends THREE.Mesh {
+    /**
+     * Finds the intersections of a ray with a mesh, assuming the mesh is a polyline extruded in
+     * the shaders (see [[SolidLineMaterial]]).
+     * @param mesh The mesh whose intersections will be found.
+     * @param raycaster Contains the intersection ray.
+     * @param intersects Array that will contain all intersections found between ray and mesh.
+     */
+    static raycast(
+        mesh: THREE.Mesh,
+        raycaster: THREE.Raycaster,
+        intersects: THREE.Intersection[]
+    ): void {
+        assert(mesh.geometry instanceof THREE.BufferGeometry, "Unsupported geometry type");
+        const geometry = mesh.geometry as THREE.BufferGeometry;
+        assert(geometry.index !== null, "Geometry does not have indices");
+        const matrixWorld = mesh.matrixWorld;
+
+        tmpInverseMatrix.getInverse(matrixWorld);
+        const localRay = tmpRay.copy(raycaster.ray).applyMatrix4(tmpInverseMatrix);
+
+        // Test intersection of ray with each of the features within the mesh.
+        if (!mesh.userData.feature) {
+            mesh.userData.feature = {};
+        }
+        if (!mesh.userData.feature.boundingVolumes) {
+            mesh.userData.feature.boundingVolumes = [];
+        }
+        const indices = geometry.index!.array;
+
+        if (Array.isArray(mesh.material)) {
+            let nextFeatureIdx = 0;
+            for (const group of geometry.groups) {
+                const material = mesh.material[group.materialIndex!];
+                const groupEndIdx = group.start + group.count;
+                nextFeatureIdx = intersectGroup(
+                    mesh,
+                    material,
+                    raycaster,
+                    localRay,
+                    intersects,
+                    nextFeatureIdx,
+                    groupEndIdx
+                );
+            }
+        } else {
+            intersectGroup(mesh, mesh.material, raycaster, localRay, intersects, 0, indices.length);
+        }
+    }
+
+    // HARP-9585: Override of base class method, however tslint doesn't recognize it as such.
+    // tslint:disable-next-line: explicit-override
```

Comment:
It is very dangerous to override the getters & setters in TS please consult with @zbigg  for insights, maybe he should also review just this part.

Reply:
done, I checked with him.